src/tools/rbd/action/Bench.cc

```diff
@@ -279,10 +298,10 @@ int do_bench(librbd::Image& image, io_type_t io_type,
       start_pos = (rand() % (size / io_size)) * io_size;
       break;
     case IO_PATTERN_SEQ:
-      start_pos = seq_chunk_length * i;
+      start_pos = (io_start + seq_chunk_length * i) % size;
```

Comment:
I don't think this handles the case when `--io-start` is set such that there is less than `--io-size` bytes available until the end of the image correctly -- `rbd_bencher_completion()` insists on being able to perform a full read at all times:
```
  } else if (b->io_type == IO_TYPE_READ && (unsigned int)ret != b->io_size) {
    std::cout << "read error: " << cpp_strerror(ret) << std::endl;
    exit(ret < 0 ? -ret : ret);
```

Reply:
Agree. There are few existing issues with IO_PATTERN_SEQ before adding io_start:

* If image size is small such that io_threads * io_size is > image size then seq_chunk_length becomes 0 and all the threads repeated issue I/O at pos 0. I think this is a bug, I'd expected that the threads would access the whole image.

* If io_bytes is greater than image size then after io_bytes of I/O each thread repeatedly issuing I/O to the same position at the start of their range. I think this is a bug, I'd expect that the positions would wrap and continue incrementing.

* If image size is not a multiple of seq_chunk_length then there is some special code to issue I/Os at the end of the image once the threads have each issued seq_chunk_length bytes of I/O. This means that I/Os are initially spaced seq_chunk_length apart, but there is then a small amount of sequential I/O at the end of the image. More a feature than a bug, I'd expected that the spacing between threads would be adjusted to deal with the remainder.

To make this work with io_start I'll make these changes:

* Add an input parameter validation constraint that io_start + io_size must be less than image size
* If io_threads * io_size > image size switch to IO_PATTERN_FULL_SEQ so that I/O occurs sequentially across the whole image
* If image size is not a multiple of seq_chunk_length then space some threads seq_chunk_length+io_size apart and the rest seq_chunk_length apart to deal with the remainder
* Fix the code so that after io_bytes has been accessed the I/O positions wrap and continue incrementing

With these changes it is then easy to adjust the start point for I/O